internal/apis/config/cainjector/v1alpha1/test/apidefaults.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	v1alpha1_pkg "github.com/cert-manager/cert-manager/internal/apis/config/cainjector/v1alpha1"
+	"github.com/cert-manager/cert-manager/pkg/apis/config/cainjector/v1alpha1"
+	"os"
+)
+
+func main() {
```

Comment:
We could reduce the complexity of this PR a bit by eg. adding a "learning mode" to the test instead of creating a seperate main.go file.
eg.
```
if os.Getenv("UPDATE_DEFAULTS") == "true" {
		if err := os.WriteFile("testdata/defaults.yaml", defaultYaml, 0644); err != nil {
			t.Fatal(err)
		}
	}
```

Reply:
Nice, yeah much tidier. Amended